xbmc/dbwrappers/dataset.h

```diff
@@ -396,10 +396,30 @@ class Dataset  {
 /* Struct to store an indexMapped field access entry */
   struct FieldIndexMapEntry
   {
-    explicit FieldIndexMapEntry(const char* name) : fieldIndex(~0), strName(name) {}
-    bool operator<(const FieldIndexMapEntry& other) const { return strName < other.strName; }
-    unsigned int fieldIndex;
-    std::string strName;
+    explicit FieldIndexMapEntry(const char* name) : fieldIndex(~0), strName(strdup(name)) {}
+
+    FieldIndexMapEntry(const FieldIndexMapEntry& other)
+      : fieldIndex(other.fieldIndex), strName(strdup(other.strName))
+    {
+    }
+
+    FieldIndexMapEntry& operator=(const FieldIndexMapEntry& other)
```

Comment:
Is a copy constructor needed? This is missing the check for autoassignment.

Reply:
Yes, it is needed. Will be called for example from `Dataset::get_index_map_entry` => `fieldIndexMap_Entries.insert(fieldIndexMap_Entries.begin() + fieldIndexMapID, tmp));`

I will add the missing self assignment check, thanks.

Reply:
On another note, we could `= delete` the move constructor/operator because we have a dynamic resource that'll leak if someone tries `std::move()`